src/ParameterDistributions.jl

```diff
@@ -873,87 +839,51 @@ function transform_constrained_to_unconstrained(
     return transf_x
 end
 
-function transform_unconstrained_to_constrained(
-    d::PDT,
-    constraints::AbstractVector,
-    x::AbstractVector{FT};
-    kwargs...,
-) where {FT <: Real, PDT <: ParameterDistributionType}
-    return cat([c.unconstrained_to_constrained(x[i]) for (i, c) in enumerate(constraints)]..., dims = 1)
-end
 
 function transform_unconstrained_to_constrained(
     d::PDT,
     constraints::AbstractVector,
-    x::AbstractMatrix{FT};
+    x::AbstractArray{FT};
     kwargs...,
 ) where {FT <: Real, PDT <: ParameterDistributionType}
-    return Array(hcat([c.unconstrained_to_constrained.(x[i, :]) for (i, c) in enumerate(constraints)]...)')
+    x_out = similar(x)
+    for (out, in, c) in zip(eachrow(x_out), eachrow(x), constraints)
+        out .= c.unconstrained_to_constrained.(in)
+    end
+    return x_out
 end
```

Comment:
much of the explanation above for `transform_constrained_to_unconstrained` applies here to. However, since `ParameterDistributionType` does not include function distributions, the size of `x_out` is the same as `x`.